src/container.ts

```diff
@@ -564,6 +566,10 @@ export function createContainer<T extends object = any, U extends object = any>(
     return resolver.resolve(container)
   }
 
+  function loadModules<ESM extends boolean = false>(
```

Comment:
Are these 2 signatures not the same? You should be able to get away with just using this one.

Reply:
Sadly not, I'm not quite sure why but if I try to use only the definition with the optional return it doesn't work claiming it doesn't match the interface:
```
Type 'Promise<{ options: ContainerOptions; cradle: T; inspect: (depth: number, opts: any) => string; cache: Map<string | symbol, CacheEntry<any>>; loadModules: <ESM extends boolean = false>(globPatterns: (string | ... 1 more ... | [...])[], opts: LoadModulesOptions<...>) => ESM extends false ? AwilixContainer<...> : Promi...' is not assignable to type 'ESM extends false ? AwilixContainer<any> : Promise<AwilixContainer<any>>'.ts(2322)
```

And of course using only the type union doesn't work since it doesn't match the condition. This suggests the same issue: https://stackoverflow.com/a/55059896/1091402